src/ot1d.jl

```diff
@@ -0,0 +1,245 @@
+"""
+    otCost1d(c, μ::UnivariateDistribution, ν::UnivariateDistribution)
+
+Calculate the optimal transport cost of the Monge-Kantorovich problem with univariate distributions `μ` and `ν` as source and target marginals and cost function `c`.
+
+Cost function `c` has to be of the form ``c(x, y) = h(|x - y|)`` where ``h`` is a
+convex function.
+"""
+function otCost1d(c, μ::UnivariateDistribution, ν::UnivariateDistribution)
```

Comment:
BTW this should be restricted to
```suggestion
function otCost1d(
    c, μ::ContinuousUnivariateDistribution, ν::ContinuousUnivariateDistribution
)
```

Reply:
Why restrict to Continuous? This still works for cases such as a Continuous Distributions and a Discrete Distribution.

Reply:
Also, if the distributions is infinitely discrete, the other code would not work, but this one would. So I think we should just leave as it was.

Reply:
In general, `otPlan1D(c, ::UnivariateDistribution, ::UnivariateDistribution)` is incorrect for finite discrete distributions and also semi-discrete optimal transport since the optimal transport map is not deterministic anymore and masses have to be split. So it seems wrong to define it for distributions that are not continuous, doesn't it?

Reply:
Actually, I think it should be `otplan1d(c, ::ContinuousDistribution, ::UniveriateDistribution)`. The only condition for the the map to exists it that `u << Lebesgue` and `c(x,y) = h(|x-y|)` for `h` convex.